docs/source-control/git-guidance/README.md

```diff
@@ -232,6 +232,74 @@ Then you can use the reflog reference (`HEAD@{}`) to reset to a specific commit
 git reset HEAD@{2}
 ```
 
+## Commit Best Practices
+
+* Make small commits. This makes changes easier to review, and if we need to revert a commit, we lose less work.
+* Commit complete and well tested code. Never commit incomplete code, get in the habit of testing your code before committing.
+* Don't mix whitespace changes with functional code changes. It is hard to determine if the line has a functional change or only removes a whitespace, so functional changes may go unnoticed.
+* Write good commit messages.
+
+## Commit Messages
+
+A commit message describes the changes made in the commit. A descriptive commit message can help understanding the code changes, and to understand why a change was made.
+A good commit message should answer these questions:
+
+* Why is it necessary? It may fix a bug, add a feature, improve performance, or just be a change for the sake of correctness
+* How does it address the issue? For short, obvious changes, this can be omitted
+* What effects does this change have? In addition to the obvious ones, this may include benchmarks, side effects etc.
+* What limitations does the current code have?
+
+Consider this when writing your commit message:
+
+* Don't assume that the code is self-evident/self-documenting
```

Comment:
Not sure I agree with this... if the code is not self-evident, we should work to make it so. 

What we write in the PR will only be there while the PR is open - the code is the long-lasting result, and as such, any questions we may have about the code, should be answered by the code itself.

Can we re-phrase this, or remove it?

Reply:
I moved this part from the original README.md
Personally, I rather say source code should be rather self-documenting then not (so I see your point here).

In fact, a commit message should rather not explain the source code but describe the change. I suggest to delete this line.